schema.graphql

```diff
@@ -1,54 +1,58 @@
-type TokenStaked @entity {
+type Deposit @entity {
   id: ID!
   by: Account!
   amount: BigInt!
+  pool: Pool!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-type TokenUnstaked @entity {
+type UnstakedDeposit @entity {
   id: ID!
   by: Account!
   amount: BigInt!
+  pool: Pool!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-type TokenStakeLockUpdated @entity {
-  id: ID!
-  by: Account!
-  depositId: BigInt!
-  lockedFrom: BigInt!
-  lockedUntil: BigInt!
-}
-
-type StakeYieldClaimed @entity {
+type Reward @entity {
   id: ID!
-  by: Account!
+  for: Account!
   amount: BigInt!
+  pool: Pool!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-type TokenPoolWeightUpdated @entity {
+type Account @entity {
   id: ID!
-  by: Account!
-  fromVal: BigInt!
-  toVal: BigInt!
+  stakes: [Deposit!]! @derivedFrom(field: "by")
+  unstakes: [UnstakedDeposit!]! @derivedFrom(field: "by")
 }
 
-type Account @entity {
+type Pool @entity {
   id: ID!
-  deposits: [Deposit!]! @derivedFrom(field: "by")
-  stakes: [TokenStaked!]! @derivedFrom(field: "by")
-  unstakes: [TokenUnstaked!]! @derivedFrom(field: "by")
+  deposits: [Deposit!]! @derivedFrom(field: "pool")
 }
 
-type Deposit @entity {
+type TokenStakeLockUpdated @entity {
   id: ID!
   by: Account!
-  index: Int!
+  depositId: BigInt!
+  lockedFrom: BigInt!
+  lockedUntil: BigInt!
   pool: Pool!
-  quantity: BigInt!
-  lockTime: BigInt!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-# How to get this information?
-type Pool @entity {
+type TokenPoolWeightUpdated @entity {
```

Comment:
Is this going to be used by anything? Just remove if it's not. Remember to consider how/why this subgraph is going to get used and only develop functionality for those use cases. Don't add things because "it might be nice"